controllers/subscription/subscription_controller.go

```diff
@@ -186,16 +188,26 @@ func (r *SubscriptionReconciler) HandleUpgrades(ctx context.Context, rhmiSubscri
 
 		return ctrl.Result{}, nil
 	}
+	log.Infof("RHMI subscription", l.Fields{"StartingCSV": rhmiSubscription.Spec.StartingCSV, "InstallPlanRef": rhmiSubscription.Status.InstallPlanRef})
+	latestRHMIInstallPlan := &olmv1alpha1.InstallPlan{}
+	err := wait.Poll(time.Second*5, time.Minute*5, func() (done bool, err error) {
+		// gets the subscription with the recreated installplan
+		err = r.Client.Get(ctx, k8sclient.ObjectKey{Name: rhmiSubscription.Name, Namespace: rhmiSubscription.Namespace}, rhmiSubscription)
+		if err != nil {
+			log.Info("Couldn't get subscription")
+			return false, nil
+		}
 
-	latestRHMIInstallPlan, err := rhmiConfigs.GetLatestInstallPlan(ctx, rhmiSubscription, r.Client)
-	if err != nil {
-		if errors.IsNotFound(err) {
-			// if installplan is not found trigger the creation of a new one
-			err = rhmiConfigs.CreateInstallPlan(ctx, rhmiSubscription, r.Client)
-			if err != nil {
-				return ctrl.Result{}, err
-			}
+		latestRHMIInstallPlan, err = rhmiConfigs.GetLatestInstallPlan(ctx, rhmiSubscription, r.Client)
+		if err != nil {
+			log.Info("Install Plan was not created")
+			return false, nil
 		}
+		return true, nil
+	})
+	if err != nil {
+		log.Info("reconcile loop")
```

Comment:
I'd change this message to say "Triggering a new reconcile loop due to an error" and the `err` variable to the log